test/e2e/tools.mjs

```diff
@@ -39,6 +42,95 @@ export async function getStats(url) {
     return fs.readJSON(path);
 }
 
+/**
+ * @param {string | URL} url
+ */
+export function getActorTestDir(url) {
+    const __filename = fileURLToPath(url);
+    const __dirname = dirname(__filename);
+    return join(__dirname, 'actor');
+}
+
+/**
+ * @param {string} dirName
+ */
+export async function runActor(dirName) {
+    let stats;
+    let datasetItems;
+
+    if (process.env.npm_config_platform) {
+        await copyPackages(dirName);
+        // TODO: add some check for token (or check that we are logged in)
+        await exec('npx -y apify-cli push', { cwd: dirName });
+
+        const actorName = await getActorName(dirName);
+        const client = new ApifyClient({ token: process.env.APIFY_TOKEN });
+        const { items: actors } = await client.actors().list();
+        const { id } = actors.find((actor) => actor.name === actorName);
+
+        const { defaultKeyValueStoreId, defaultDatasetId } = await client.actor(id).call();
```

Comment:
it waits indefinitely unless some value is provided, yes.

So I replaced `new ApifyClient({ token: process.env.APIFY_TOKEN });` with `Actor.newClient()`, but otherwise kept the rest to go through the client to kinda not mix Actor call and client calls + I think with `Actor.openKeyValueStore()`, `kvStore.getValue()` it would be less straight forward for someone who would see it for the first time.